fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -349,11 +353,11 @@ private static void timerIncrement(@Nullable FDBStoreTimer timer, FDBStoreTimer.
     }
 
     protected <T> CompletableFuture<Void> iterateRangeOnly(@Nonnull FDBRecordStore store,
-                                                           @Nonnull RecordCursor<T> cursor,
+                                                           @Nonnull RecordCursor<T> cursor, // at least two items must be requested for this cursor
```

Comment:
Oh, it was suppose to mean that the cursor should be called with a minimal limit of 2 items (else it'll be an infinite loop). 
The cursor itself, though, may have any number of items (including 0 and 1). 
I'll replace this comment with one that clarify how this function returns the last cursor's item for continuation. This will be more helpful, I think.